### ISSUE TITLE
Julia just exit without leaving any error message in the middle of package installation

### ISSUE THREAD

**Issue by eisthf** [JuliaLang/Pkg.jl#1857]
All of sudden, I can't install any package.
If I try installing any package, Julia just silently exit in the middle without leaving any error message.
I have no idea if it is Pkg.jl or Julia that makes the trouble. So I filed the same error report to Julia (https://github.com/JuliaLang/julia/issues/36195)

The following is all I see from the terminal.
```
E:\>julia
(@v1.4) pkg> add CSV
   Updating registry at `E:\.julia\registries\General`
   Updating git-repo `https://github.com/JuliaRegistries/General.git`
  Resolving package versions...
```

I uninstalled Julia, deleted everything under .julia and restarted but the problem persists.

I tried other builds and found pkg installation is ok only with Julia 1.0.5. Other versions after that such as 1.3.1, 1.4.0, 1.4.1, 1.4.2, 1.5.0-beta1 all fail.

It used to work well until a few days ago. It seems the issue occurs only on two Windows PCs that I use at the company office (no problem on Linux). Since these PCs' system configurations are remotely managed by IT department time to time, I suspect somehow some recent changes are conflicting with Julia; Pkg install works well on Windows PC at home. Or it may be related to Windows update?

Here is my environment (since REPL or Pkg does not produce any output I can't give further detail):
```
julia> versioninfo()
Julia Version 1.4.2
Commit 44fa15b150* (2020-05-23 18:35 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, skylake)
Environment:
  JULIA_DEPOT_PATH = E:\.julia
  JULIA_NUM_THREADS = 6
```

===================
Update:

The reason of failure is that `symlink()` fails inside `probe_symlink_creation()` call.

Indeed, if I just run the following, `symlink()` call fails.
```
julia> symlink("foo", "D:/Cygwin/tmp/0fCzeUp3zgpj")
ERROR: IOError: symlink: permission denied (EACCES)
```
What's strange is that `mklink texput-sym.log texput.log` works ok in Windows cmd with Admin privilege. 

Initially I thought I found a work-around for the issue by just returning `true` in `probe_symlink_creation()` but it worked only for a few packages such as CSV.

DataFramesMeta fails to get installed even with the above approach. Frustrating...

===============
Update:
`ENV["BINARYPROVIDER_COPYDEREF"]="true"` solves the problem. It is only mentioned in the source file. It would be great if it is also mentioned in the document. I would like to keep this open for a couple of days just in case.

**Comment by staticfloat** [comment]
@eisthf what version of Julia are you using?  Julia closing with no error message sounds like a segfault.

**Comment by eisthf** [comment]
@staticfloat I tried 1.3.1, 1.4.0, 1.4.1, 1.4.2, 1.5.0-beta1. All fail for the same reason: symlink() kind of aborts inside probe_symlink_creation().

What's strange is that the symptom I am experiencing never happened until recently. I was able to install packages without any problem.

Currently I set ENV["BINARYPROVIDER_COPYDEREF"]="true" in startup.jl and all is well with that setting. I suspect there must be a downside of the setting though. I hope it is not serious.
